pyuvdata/uvdata/mwa_corr_fits.py

```diff
@@ -741,15 +770,15 @@ def read_mwa_corr_fits(
                 # by a factor of 64 here. For a more detailed explanation, see PR #908.
                 dig_gains = dig_gains[:, coarse_inds] / 64
                 dig_gains = np.repeat(dig_gains, num_fine_chans, axis=1)
-                dig_gains1 = dig_gains[self.ant_1_array, :]
-                dig_gains2 = dig_gains[self.ant_2_array, :]
-                dig_gains1 = dig_gains1[:, :, np.newaxis]
-                dig_gains2 = dig_gains2[:, :, np.newaxis]
-                if self.data_array.dtype != np.complex128:
-                    self.data_array = self.data_array.astype(np.complex128)
-                self.data_array = self.data_array / (dig_gains1 * dig_gains2)
+
+                self.data_array /= (
+                    dig_gains[self.ant_1_array, :, np.newaxis]
+                    * dig_gains[self.ant_2_array, :, np.newaxis]
+                )
+
                 if self.data_array.dtype != data_array_dtype:
```

Comment:
@mkolopanis I think the data type checking here (and the other one below) are not needed anymore. The initial declaration happens using the `data_array_dtype` argument, and the rest of the manipulations involve array reshaping and/or inplace updates. If we wanted to be paranoid we could change these to `assert` statements, but I'm also okay with removing them.

Reply:
I went ahead and removed these statements while I was adding the new test.